tests/tests.py

```diff
@@ -938,3 +939,228 @@ def test_invalidate_model_with_geometry(self):
         geom.save()
         # Raises ValueError if this doesn't work
         invalidate_obj(geom)
+
+
+class IntentionalRollback(RuntimeError):
+    pass
+
+
+class ThreadWithReturnValue(Thread):
+    def __init__(self, *args, **kwargs):
+        super(ThreadWithReturnValue, self).__init__(*args, **kwargs)
+        self._return = None
+
+    def run(self):
+        try:
+            # python 3
+            if self._target is not None:
+                self._return = self._target(*self._args, **self._kwargs)
+        except AttributeError:
+            # python 2
+            if self._Thread__target is not None:
+                self._return = self._Thread__target(*self._Thread__args, **self._Thread__kwargs)
+
+    def join(self, *args, **kwargs):
+        super(ThreadWithReturnValue, self).join(*args, **kwargs)
+        return self._return
+
+
+def get_category_filter_pk_1():
+    try:
+        return list(Category.objects.filter(pk=1).cache())
+    finally:
+        from django.db import connection
+        connection.close()
```

Comment:
I was getting connections left open sometimes in Postgres when the test went to delete the CB. Manual closing was recommended in a django issue. I'll see if I can re find the link and put it in the source.

Joel

> On Oct 16, 2015, at 19:31, Alexander Schepanovski <notifications@github.com> wrote:
> 
> In tests/tests.py:
> 
> > +        except AttributeError:
> > +            # python 2
> > +            if self._Thread__target is not None:
> > +                self._return = self._Thread__target(*self._Thread__args, **self._Thread__kwargs)
> > +
> > +    def join(self, *args, **kwargs):
> > +        super(ThreadWithReturnValue, self).join(*args, **kwargs)
> > +        return self._return
> > +
> > +
> > +def get_category_filter_pk_1():
> > +    try:
> > +        return list(Category.objects.filter(pk=1).cache())
> > +    finally:
> > +        from django.db import connection
> > +        connection.close()
> Why explicit .close()?
> 
> ―
> Reply to this email directly or view it on GitHub.
>